EIPS/eip-7702.md

```diff
@@ -23,63 +23,126 @@ There is a lot of interest in adding short-term functionality improvements to EO
+
+authorization_list = [[chain_id, address, nonce, y_parity, r, s], ...]
 ```
 
-The intrinsic cost of the new transaction is inherited from [EIP-2930](./eip-2930.md), specifically `21000 + 16 * non-zero calldata bytes + 4 * zero calldata bytes + 1900 * access list storage key count + 2400 * access list address count`. Additionally, we add a cost of `16 * non-zero calldata bytes + 4 * zero calldata bytes` over each `contract_code`, plus `PER_CONTRACT_CODE_BASE_COST` times the length of the `contract_code` array.
+The fields `chain_id`, `nonce`, `max_priority_fee_per_gas`, `max_fee_per_gas`, `gas_limit`, `destination`, `value`, `data`, and `access_list` of the outer transaction follow the same semantics as [EIP-1559](./eip-1559.md).
+
+The `authorization_list` is a list of tuples that store the address to code which the signer desires to set in their EOA temporarily. Nonce is considered optional.
+
+The [EIP-2718](./eip-2718.md) `ReceiptPayload` for this transaction is `rlp([status, cumulative_transaction_gas_used, logs_bloom, logs])`.
+
+#### Behavior
+
+At the start of executing the transaction, for each `[chain_id, address, nonce, y_parity, r, s]` tuple:
+
+1. `authority = ecrecover(keccak(MAGIC || rlp([chain_id, nonce, address])), y_parity, r, s]`
+2. Verify the chain id is either 0 or the chain's current ID.
+3. Verify that the code of `authority` is empty.
+4. If nonce is not null, verify the nonce of `authority` is equal to `nonce`.
+5. Set the code of `authority` to code associated with `address`.
+6. Add the `authority` account to `accessed_addresses` (as defined in [EIP-2929](./eip-2929.md).)
+
+If any of the above steps fail, immediately stop processing that tuple and continue to the next tuple in the list. It will In the case of multiple tuples for the same authority, set the code specified by the address in the first occurrence.
+
+At the end of the transaction, set the code of each `authority` back to empty.
 
-At the start of executing the transaction, for each `[contract_code, y_parity, r, s]` tuple:
+Note that the signer of an authorization tuple may be different than `tx.origin` of the transaction.
 
-1. Let `signer = ecrecover(keccak(MAGIC + contract_code), y_parity, r, s]`.
-2. Verify that the contract code of `signer` is empty.
-3. Set the contract code of `signer` to `contract_code`.
-4. Add the `signer` account to `accessed_addresses` (as defined in [EIP-2929](./eip-2929.md).)
+#### Gas Costs
 
-At the end of the transaction, set the `contract_code` of each `signer` back to empty.
+The intrinsic cost of the new transaction is inherited from [EIP-2930](./eip-2930.md), specifically `21000 + 16 * non-zero calldata bytes + 4 * zero calldata bytes + 1900 * access list storage key count + 2400 * access list address count`. Additionally, we add a cost of `PER_CONTRACT_CODE_BASE_COST * authorization list length`.
```

Comment:
suggesting per-contract code item calldata and the wrapper tx calldata is expected? where is the former supposed to be included and what is the point of the latter if we have the former?